File: domain/chat_model.py
from typing import Dict

# Custom exception for domain-related errors
class DomainError(Exception):
    """Custom exception for domain errors in the domain layer."""
    pass

class ChatRequest:
    def __init__(self, message: str) -> None:
        if not message:
            raise DomainError("Message cannot be empty.")
        self.message: str = message

    def to_dict(self) -> Dict[str, str]:
        """Convert ChatRequest to a dictionary."""
        return {"message": self.message}

class ChatResponse:
    def __init__(self, user_message: str, bot_response: str) -> None:
        if not user_message or not bot_response:
            raise DomainError("Both user_message and bot_response are required.")
        self.user_message: str = user_message
        self.bot_response: str = bot_response

    def to_dict(self) -> Dict[str, str]:
        """Convert ChatResponse to a dictionary."""
        return {
            "user_message": self.user_message,
            "bot_response": self.bot_response
        }

File: domain/chat_model.ts
export class ChatRequest {
    message: string;
  
    constructor(message: string) {
      if (!message) {
        throw new Error("Message cannot be empty.");
      }
      this.message = message;
    }
  
    toDict(): { message: string } {
      return { message: this.message };
    }
  }
  
  export class ChatResponse {
    userMessage: string;
    botResponse: string;
  
    constructor(userMessage: string, botResponse: string) {
      if (!userMessage || !botResponse) {
        throw new Error("Both userMessage and botResponse are required.");
      }
      this.userMessage = userMessage;
      this.botResponse = botResponse;
    }
  
    toDict(): { user_message: string; bot_response: string } {
      return {
        user_message: this.userMessage,
        bot_response: this.botResponse
      };
    }
  }
  